anesthetic/plot.py

```diff
@@ -402,6 +392,57 @@ def contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
     return cbar
 
 
+def hist_plot_2d(ax, data_x, data_y, *args, **kwargs):
+    """Plot a 2d marginalised distribution as a histogram.
+
+    This functions as a wrapper around matplotlib.axes.Axes.hist2d
+
+    Parameters
+    ----------
+    ax: matplotlib.axes.Axes
+        axis object to plot on
+
+    data_x, data_y: numpy.array
+        x and y coordinates of uniformly weighted samples to generate kernel
+        density estimator.
+
+    xmin, xmax, ymin, ymax: float
+        lower/upper prior bounds in x/y coordinates
+        optional, default None
+
+    Returns
+    -------
+    c: matplotlib.collections.QuadMesh
+        A set of colors
+
+    """
+    xmin = kwargs.pop('xmin', None)
+    xmax = kwargs.pop('xmax', None)
+    ymin = kwargs.pop('ymin', None)
+    ymax = kwargs.pop('ymax', None)
+    label = kwargs.pop('label', None)
+    color = kwargs.pop('color', next(ax._get_lines.prop_cycler)['color'])
+
+    if len(data_x) == 0 or len(data_y) == 0:
+        return numpy.zeros(0), numpy.zeros(0), numpy.zeros((0, 0))
+
+    pdf, x, y = numpy.histogram2d(data_x, data_y, density=True)
+    contours = iso_probability_contours(pdf)
+    pdf = numpy.digitize(pdf, contours, right=True)
+    pdf = numpy.array(contours)[pdf]
+    pdf = numpy.ma.masked_array(pdf, pdf < contours[1])
+    cmap = basic_cmap(color)
+    image = ax.pcolormesh(x, y, pdf.T, cmap=cmap, vmin=0, vmax=pdf.max(),
```

Comment:
Why `pcolormesh` here instead of `hist2d`? 
`hist2d` is the standard 2d version of the 2d `hist` with with mostly the same kwargs (e.g. `bins`, `range`).

Reply:
It depends on whether you want histograms that look like this:

![hist1](https://user-images.githubusercontent.com/3708486/62188311-dfd62b80-b363-11e9-97fa-e7591af694b1.png)

or like this

![hist2](https://user-images.githubusercontent.com/3708486/62188325-e8c6fd00-b363-11e9-889d-3dc660476ffd.png)

If we want the same contouring effect as we have for kernel density estimation, then we have to split these up. However, the latter simple wrapper around `plt.hist2D` is considerably simpler than the above implementation. We could possibly include an option for both?

Reply:
9cee366 Addresses this properly, both in providing a complete `hist2d` option, allowing users to provide contours, and also better mimicking the
[source code](https://matplotlib.org/3.1.1/_modules/matplotlib/axes/_axes.html#Axes.hist2d) of `hist2d`. This means that `bins`, `range` etc can now be passed to `MCMCSamples.hist_plot_2d`